compiler/rustc_hir_typeck/src/closure.rs

```diff
@@ -80,7 +80,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         debug!(?bound_sig, ?liberated_sig);
 
-        let mut fcx = FnCtxt::new(self, self.param_env.without_const(), body.value.hir_id);
+        let body_def_id = self.tcx.hir().local_def_id(body.value.hir_id);
+        let mut fcx = FnCtxt::new(self, self.param_env.without_const(), body_def_id);
```

Comment:
looks like that this will ICE

Reply:
I think you'll want to use `closure.def_id` here?

Reply:
Oh nice, yeah this should work!